File: src/backend/app/models/Office.py
# Office.py

from neomodel import StringProperty, RelationshipFrom, RelationshipTo
from .BaseModel import BaseModel

class Office(BaseModel):
    name = StringProperty(max_length=60)
    address = StringProperty(max_length=120)
    email = StringProperty()
    phone_number = StringProperty()
    opening_hours = StringProperty(max_length=60)
    meta = StringProperty(max_length=300)

    # Связи
    doctors = RelationshipFrom('app.models.Doctor', 'WORKS_IN')            # В офисе работают врачи
    appointments = RelationshipFrom('app.models.Appointment', 'PROVIDES')  # В офисе проходят приёмы
    procedures = RelationshipFrom('app.models.Procedure', 'PROVIDES')      # В офисе предоставляются процедуры

File: src/frontend/models/Office.js
export class Office {
    constructor(name, address, email, phone_number, opening_hours, meta, doctors = [], appointments = [], procedures = []) {
        this.name = name;
        this.address = address;
        this.email = email;
        this.phone_number = phone_number;
        this.opening_hours = opening_hours;
        this.meta = meta;
        this.doctors = doctors;
        this.appointments = appointments;
        this.procedures = procedures;
    }
}
